cmd/restic/cmd_init.go

```diff
@@ -69,18 +85,54 @@ func runInit(opts InitOptions, gopts GlobalOptions, args []string) error {
 
 	err = s.Init(gopts.ctx, gopts.password, chunkerPolynomial)
 	if err != nil {
-		return errors.Fatalf("create key in repository at %s failed: %v\n", location.StripPassword(gopts.Repo), err)
+		if !gopts.JSON {
+			return errors.Fatalf("create key in repository at %s failed: %v\n", location.StripPassword(gopts.Repo), err)
+		} else {
+			status := struct {
+				Status     string `json:"status"`
+				Repository string `json:"repository"`
+				Message    string `json:"message"`
+			}{
+				Status:     "error_key",
+				Repository: location.StripPassword(gopts.Repo),
+				Message:    err.Error(),
+			}
+			return errors.Fatalf(toJSONString(status))
+		}
 	}
 
-	Verbosef("created restic repository %v at %s\n", s.Config().ID[:10], location.StripPassword(gopts.Repo))
-	Verbosef("\n")
-	Verbosef("Please note that knowledge of your password is required to access\n")
-	Verbosef("the repository. Losing your password means that your data is\n")
-	Verbosef("irrecoverably lost.\n")
+	if !gopts.JSON {
+		Verbosef("created restic repository %v at %s\n", s.Config().ID[:10], location.StripPassword(gopts.Repo))
+		Verbosef("\n")
+		Verbosef("Please note that knowledge of your password is required to access\n")
+		Verbosef("the repository. Losing your password means that your data is\n")
+		Verbosef("irrecoverably lost.\n")
+
+	} else {
+		status := struct {
+			Status     string `json:"status"`
+			Id         string `json:"id"`
```

Comment:
The standard capitalization for `ID` of variables names in restic is all caps (The JSON field name is fine).